packages/sync/src/createClient.ts

```diff
@@ -21,20 +22,30 @@ export const createClient = async (userConfig?: DataSyncConfig): Promise<Voyager
   const clientConfig = extractConfig(userConfig);
   const { cache } = await buildCachePersistence(clientConfig);
 
-  const link = await defaultLink(clientConfig);
+  const offlineLink = await createOfflineLink(clientConfig);
```

Comment:
This gives a lot of flexibility now however we are no longer have all logic in links. 
restoreOfflineOperations functions are external and require access to the link.